datacube/scripts/dataset.py

```diff
@@ -462,6 +463,28 @@ def _get_derived_set(index: Index, id_: str) -> Set[Dataset]:
     return derived_set
 
 
+@dataset_cmd.command('uri-search')
+@click.option('--search-mode', help='Exact, prefix or guess based searching',
+              type=click.Choice(['exact', 'prefix', 'guess']), default='prefix')
+@click.argument('paths', nargs=-1)
+@ui.pass_index()
+def uri_search_cmd(index: Index, paths: List[str], search_mode):
+    """
+    Search by dataset locations
+
+    PATHS may be either file paths or URIs
+    """
+    if search_mode == 'guess':
+        # This is what the API expects. I think it should be changed.
+        search_mode = None
+    for path in paths:
+        dss = list(index.datasets.get_datasets_for_location(uri_resolve(base=path), mode=search_mode))
```

Comment:
Shorthand `dss` is less readable, but we do it everywhere...

Reply:
:(
It can stay consistent for now, and maybe get changed everywhere later.